src/orient/rational-calc.orient

```diff
@@ -0,0 +1,704 @@
+        describe(post_mtree_layers_cached, "Number of layers of the replica merkle tree being cached")
+        describe(post_mtree_layers_deleted, "Number of layers of the replica merkle tree not being cached")
+
+        post_mtree_cached = post_mtree_cached_hashnodes * node_size
+        post_mtree_cached_gib = post_mtree_cached / b_in_gib 
+        post_mtree_overhead = post_mtree_cached / sector_size
+        post_mtree_cached_hashnodes = expt(2, post_mtree_layers_cached) - 1
+        post_mtree_layers_deleted = commr_tree_depth - post_mtree_layers_cached
+
+      Generation:
+        describe(post_mtree_regen_hashnodes, "Number of intermediary nodes to be regenerated for a single replica merkle tree inclusion proof in PoSt")
+        describe(post_mtree_regen_nodes, "Number of leaf nodes to be access for a single replica merkle tree inclusion proof in PoSt")
+        describe(post_inclusion_time, "Time to generate a Merkle tree inclusion proof for PoSt (assuming all data in memory)", secs)
+
+        post_mtree_regen_hashnodes = expt(2, post_mtree_layers_deleted + 1) - 1 // TODO check if -1 should be removed
+
+        post_mtree_regen_nodes = expt(2, post_mtree_layers_deleted) - 1 // TODO check if -1 should be removed
+        post_inclusions_time = post_inclusion_time * post_challenges
+        post_inclusions_time_parallel = post_inclusions_time / rig_cores
+
+      Generation (Wrapping):
+        post_inclusion_time = (post_mtree_regen_hashnodes * 2) * merkle_tree_hash_time
+
+      Generation (StackedReplicas):
+        post_inclusion_time = (post_mtree_regen_hashnodes * 2) * merkle_tree_hash_time
+
+      Generation (WrappingVariant):
+        describe(wrapper_kdf_time, "Time to generate a node on the wrapper layer", secs)
+
+        wrapper_kdf_time = kdf_time * wrapper_parents_all * post_mtree_regen_nodes
+        post_inclusion_time = (post_mtree_regen_hashnodes * 2) * merkle_tree_hash_time + wrapper_kdf_time
+
+  StorageAccess:
+    describe(post_challenge_read, "Time to read from storage medium data required for a single challenge in PoSt", secs)
+    describe(post_data_access, "Time to retrieve from the storage unit the data for PoSt", secs)
+    describe(post_data_access_parallel, "Time to retrieve from the storage unit the data in parallel for PoSt", secs)
+    describe(post_leaf_read, "Time to retrieve the challenged leaf in PoSt", secs)
+    describe(post_leaves_read, "Time to retrieve the all the challenged leaves in PoSt", secs)
+    describe(post_leaves_read_parallel, "Time to retrieve the all the challenged leaves in PoSt in parallel", secs)
+
+    post_data_access = post_challenge_read * post_challenges
+    post_leaves_read = post_leaf_read * post_challenges
+    // TODO(enhancement): calculate first mtree and then challenge_read
+    post_mtree_read = (post_challenge_read - post_leaf_read) * post_challenges
```

Comment:
The magnitudes being subtracted were already multiplied by `post_challenges`, I'm not finding the definition of this variable but this seems suspicious.